node/src/components/consensus/highway_core/state.rs

```diff
@@ -551,6 +559,26 @@ impl<C: Context> State<C> {
                 }
             }
         }
+        // All endorsed units from the panorama of this vote.s
+        let endorsements_in_panorama: Vec<C::Hash> =
+            panorama
+                .iter_correct_hashes()
+                .fold(Vec::new(), |mut acc, hash| {
+                    acc.extend(
+                        self.opt_endorsements(hash)
+                            .unwrap_or_default()
+                            .into_iter()
+                            .map(|e| *e.unit())
+                            .collect::<HashSet<_>>(),
+                    );
+                    acc
+                });
```

Comment:
Why accumulate all of that in a `Vec`? Why not just `flat_map`? No collecting or allocating should be needed here, except maybe to deduplicate?

Reply:
:grimacing:

Reply:
Done.